tls/s2n_handshake_io.c

```diff
@@ -1108,6 +1108,16 @@ int s2n_conn_set_handshake_no_client_cert(struct s2n_connection *conn)
     return S2N_SUCCESS;
 }
 
+S2N_RESULT s2n_conn_record_state_machine(struct s2n_connection *conn, bool use_tls13)
+{
+    RESULT_ENSURE_REF(conn);
+
+    conn->handshake.tls13_state_machine = use_tls13;
+    conn->handshake.state_machine_fixed = true;
```

Comment:
Would it make more sense to use an enum here? S2N_STATE_MACHINE_INITIAL, S2N_STATE_MACHINE_TLS12, S2N_STATE_MACHINE_TLS13? I know it does take more memory than two bits, but not much and it might be clearer.